array.c

```diff
@@ -4373,65 +4373,81 @@ take_items(VALUE obj, long n)
 
 /*
  *  call-seq:
- *    array.zip(*other_arrays) -> new_array
- *    array.zip(*other_arrays) {|other_array| ... } -> nil
+ *    zip(*other_arrays) -> new_array
+ *    zip(*other_arrays) {|other_array| ... } -> nil
  *
- *  When no block given, returns a new +Array+ +new_array+ of size <tt>self.size</tt>
- *  whose elements are Arrays.
+ *  With no block given, combines +self+ with the collection of +other_arrays+;
+ *  returns a new array of sub-arrays.
  *
- *  Each nested array <tt>new_array[n]</tt> is of size <tt>other_arrays.size+1</tt>,
- *  and contains:
+ *  Returns an array of size <tt>other_arrays.size + 1</tt>
```

Comment:
I don't think this is true? It returns an array of size `self.size` and each subarray is of size `other_arrays.size + 1`:

```ruby
[1, 2].zip([3, 4], [5, 6], [7, 8]) # => [[1, 3, 5, 7], [2, 4, 6, 8]]
```

Reply:
Changed.